### ISSUE TITLE
Make sure SQL sandbox ACS works with static time

### ISSUE THREAD

**Issue by rautenrieth-da** [digital-asset/daml#150]
Currently, the contracts table stores creation and archival timestamps (see [SQL schema](https://github.com/DACH-NY/daml/blob/master/ledger/sandbox/src/main/resources/db/migration/V1__Init.sql#L50)). This is used to get the active contract set at a given offset. However, the mapping from offset to timestamp is not injective (e.g., for static time mode).

The schema should be changed to store creation and archival transaction IDs instead.

@francesco-da

**Comment by stefanobaghino-da** [comment]
@rautenrieth-da Is this issue still valid? I have a feeling the current tests seem to prove that the ACS works with static time.

**Comment by rautenrieth-da** [comment]
The ACS was fixed some time ago to use offsets to decide whether the contract is active.

The second question referred to the fact that LET and not offset is used to validate transactions in https://github.com/digital-asset/daml/blob/master/ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ActiveLedgerStateManager.scala#L106. I was worried about scenarios like the following:
- 2 transactions with the same LET get serialized in parallel
- The "second" one gets committed fast and creates a contract X
- The "first" one takes longer to serialize. By the time it gets to the above check, contract X is already in the ACS and (assuming _Read committed_ isolation level) it is decided that the reference is valid.
- Now you have a "first" transaction that refers to a contract created in a "second" transaction.

@gerolf-da

**Comment by rautenrieth-da** [comment]
Closing in favor of specific issues #3543 and #3544.
